pkg/builder/controller.go

```diff
@@ -145,6 +149,12 @@ func (blder *Builder) WithOptions(options controller.Options) *Builder {
 	return blder
 }
 
+// WithLogger overrides the controller options's logger used. Defaults to nil logger.
```

Comment:
It might be nice to extend this same type of Options helper for setting the logger to `manager.New()` as well, I'm not sure it would completely eliminate the need to initialize a logger with `ctrl.SetLogger`, but might be a good start down that path.

It would also allow a way for controllers built with `ctrl.NewControllerManagedBy(mgr)` to inherit the same base logger without having to configure it in multiple places or rely on `ctrl.SetLogger()` for the behavior.

Reply:
Done, ptal

Reply:
lgtm, thanks!